src/channel.js

```diff
@@ -21,13 +21,18 @@ export function Channels(channels, data) {
 }
 
 // TODO Use Float64Array for scales with numeric ranges, e.g. position?
+// TODO Cleaner way of exposing “raw” (unscaled) values?
 export function valueObject(channels, scales) {
-  return Object.fromEntries(
+  const raw = {};
+  const values = Object.fromEntries(
     Object.entries(channels).map(([name, {scale: scaleName, value}]) => {
       const scale = scaleName == null ? null : scales[scaleName];
+      raw[name] = value;
       return [name, scale == null ? value : map(value, scale)];
     })
   );
+  values.raw = raw;
+  return values;
 }
```

Comment:
Here we'll need to take into account that the scaled values can diverge from their original position when going through an initializer. They might even not be based on the same index or have the same cardinality as the unscaled channels.

For example, if you try:
```
Plot.tooltip(athletes, Plot.hexbin(
  {fill:"count"},
  {x: "weight", y: "height", fx: "sex", fy: "date_of_birth"}
))
```

the raw channels are:
`{x: Array(242), y: Array(242), fill: Float64Array(242), fx: Array(11538), fy: Array(11538)}`

x and y contain scaled values, so they're not "raw"; and fy and fy are not parallel to the new index.

Just flagging—I have no real idea on how to solve this atm.

Reply:
I think the best we can do is that the channels represent the transformed data. In the example you gave, it should be equivalent to precomputing the hexagonal bins prior to calling Plot.plot and passing them in. Therefore I think that the _fx_ and _fy_ channels are the ones that are “wrong” as they don’t match the (transformed, aggregated) data and index. (We could try to devise a formal mechanism to allow transforms to describe aggregations, but I think that will be tricky and it doesn’t generalize that well, for example with weighted rolling window transforms.)

Since the hexbin transform operates in screen space, this implies that there wouldn’t be an automatic way to show the abstract _x_ and _y_ values of the hexbin that you are hovering over. But that _x_ and _y_ merely represent the centroid of the hexbin anyway, so it’s not particularly meaningful (especially if _x_ or _y_ is not linear), and if you’re binning on _x_ and _y_, it implies you don’t care about the precise value of _x_ and _y_ anyway. More likely what you want in that case is for a tooltip to show other details, _e.g._, the count of athletes in that particular hexagon, say, or maybe the distribution of their nationalities or sports.

Reply:
I dropped the _fx_ and _fy_ channels in 0486d24. (We could impute _fx_ and _fy_ channels if we wanted, but that would just be slower so it seems cleaner this way.)